feat(steamcmd): set platformtype for steamcmd in the default config

h3o66 opened the pull request:
# Description

Refactors the existing code to remove the hardcoding in the core_dl for steamcmd for the gameservers

## Type of change

* [ ] Bug fix (a change which fixes an issue).
* [ ] New feature (change which adds functionality).
* [ ] New Server (new server added).
* [x] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

* [ ] This pull request links to an issue.
* [x] This pull request uses the `develop` branch as its base.
* [x] This pull request Subject follows the Conventional Commits standard.
* [x] This code follows the style guidelines of this project.
* [x] I have performed a self-review of my code.
* [x] I have checked that this code is commented where required.
* [x] I have provided a detailed with enough description of this PR.
* [ ] I have checked If documentation needs updating.

## Documentation

If documentation does need updating either update it by creating a PR (preferred) or request a documentation update.
* User docs: https://github.com/GameServerManagers/LinuxGSM-Docs
* Dev docs: https://github.com/GameServerManagers/LinuxGSM-Dev-Docs

**Thank you for your Pull Request!**

dgibbs64 reviewed (changes requested):
needs to be a yes or no rather than 1 or 0 to match the currently convention

h3o66 commented:
> needs to be a yes or no rather than 1 or 0 to match the currently convention

Changed this with the latest commit and rebased the banch on develop

dgibbs64 reviewed (approved):
New servers recently added to develop. Please make sure they are added to
